### PR TITLE
[WIP] fixed magnetometer detection

### PR DESCRIPTION
**Pull request by jerelbn**

### CODE REVIEW

File: src/sensors.c [hunk 1]
```diff
@@ -90,27 +90,38 @@ bool update_sensors()
...
+          _diff_pressure_present = ms4525_init();
+          if(_diff_pressure_present)
+          {
+            mavlink_log_info("FOUND DIFF PRESS", NULL);
+          }
+        }
+        if(!_mag_present)
         {
-          mavlink_log_info("FOUND DIFF PRESS", NULL);
+          _mag_present = hmc5883lInit(get_param_int(PARAM_BOARD_REVISION));
```
Comment (superjax) [inline src/sensors.c]: Do you know how long this function takes if the magnetometer is not present?

Comment (jerelbn) [inline src/sensors.c]: It looks like there are 10ms delays in two differen for loops for a total of 100ms of delay. Do I need the debugger to time the hmc5883lInit function?

Comment (superjax) [inline src/sensors.c]: No, I'll show you tomorrow how to use the `mavlink_log_info` functions to print to the screen from the naze

**Comment by superjax** [comment]
Let's wait until the magnetometer calibration is finished before merging this in.

**Comment by jerelbn** [comment]
the magnetometer calibration is complete on ROSflight and on the fcu_io mag_cal branch

**Comment by superjax** [comment]
Hey, I just tried this out, and it looks great! 

I have one question though, what units are these once it's calibrated?  Mine seem really really large, I expected them to be in gauss, which would result in something like 0.65 gauss.  I just did a calibration, and I'm getting something like these:

```
magnetic_field: 
  x: 13860.0888672
  y: -48064.234375
  z: 11638.2470703
```

Here is the results from my calibration
```
[ INFO] [1488672144.068187837]: Magnetometer calibration complete:
A = 
100.16690,   1.41608,   1.22894,
  1.41608,  97.54987,   0.04498,
  1.22894,   0.04498, 100.56349
bx = 145.03518, by =  39.85698, bz = -188.19234
[ WARN] [1488672144.068289723]: Write params to save new magnetometer calibration!
[ INFO] [1488672144.072125863]: Parameter MAG_A11_COMP has new value 100.167
[ INFO] [1488672144.072557846]: Parameter MAG_A12_COMP has new value 1.41608
[ INFO] [1488672144.072589212]: Parameter MAG_A13_COMP has new value 1.22894
[ INFO] [1488672144.072621644]: Parameter MAG_A21_COMP has new value 1.41608
[ INFO] [1488672144.072661384]: Parameter MAG_A22_COMP has new value 97.5499
[ INFO] [1488672144.075806766]: Parameter MAG_A23_COMP has new value 0.0449756
[ INFO] [1488672144.075861083]: Parameter MAG_A31_COMP has new value 1.22894
[ INFO] [1488672144.075890037]: Parameter MAG_A32_COMP has new value 0.0449756
[ INFO] [1488672144.075917509]: Parameter MAG_A33_COMP has new value 100.563
[ INFO] [1488672144.076105283]: Parameter MAG_X_BIAS has new value 145.035
[ INFO] [1488672144.076143033]: Parameter MAG_Y_BIAS has new value 39.857
[ INFO] [1488672144.076174960]: Parameter MAG_Z_BIAS has new value -188.192
```

Any ideas?

**Comment by jerelbn** [comment]
I scaled it to nano Tesla, which is about 51000 in Provo. I saw those unit in lots of places, so that's why I picked it, but we can scale it to whatever units as long as it's specified in the code. I don't think it is specified right now.

**Comment by superjax** [comment]
It really doesn't matter to me.  nT sounds fine, let's just specify that in the code.

**Pull request merged**